SpotifyUserLogin/src/main/java/useCase/SurpriseMeInteractor.java

```diff
@@ -0,0 +1,48 @@
+package useCase;
+
+import api.SpotifyInteractor;
+import entities.Song;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.Random;
+
+public class SurpriseMeInteractor {
+    private final SpotifyInteractor spotifyInteractor;
+
+    public SurpriseMeInteractor(SpotifyInteractor spotifyInteractor) {
+        this.spotifyInteractor = spotifyInteractor;
+    }
+
+    public Song getRandomSongFromFriends(boolean usePreferences) {
```

Comment:
Comments aren't bad but I would still recommend splitting this into helper methods